packages/dds/map/src/directory.ts

```diff
@@ -980,21 +986,18 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
     /**
      * {@inheritDoc IDirectory.get}
      */
-    public get<T = any>(key: string): T {
-        if (!this._storage.has(key)) {
-            return undefined;
-        }
-
-        return this._storage.get(key).value as T;
+    public get<T = any>(key: string): T | undefined {
+        return this._storage.get(key)?.value as T | undefined;
     }
 
     /**
      * {@inheritDoc IDirectory.wait}
      */
-    public async wait<T = any>(key: string): Promise<T> {
+    public async wait<T = any>(key: string): Promise<T | undefined> {
```

Comment:
This should only be `T` right?  Since the promise just won't resolve if the key isn't set.

Reply:
Yeah... I was debating in my head, since we are wait is listening on `valueChanged` and the `delete` action would also cause it.  But send we already check if doesn't exist up front, then only a `set` action would cause the `valueChanged`, and that means that we should just follow the user's type assertion here.  I will change.